### PR TITLE
Add javadoc to ConsoleIO

### PR DESCRIPTION
**Pull request by joeyfezster**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by joeyfezster** [comment]
@staslev Please review

**Comment by joeyfezster** [comment]
Retest this please

**Comment by staslev** [comment]
Retest this please

File: runners/spark/src/main/java/org/apache/beam/runners/spark/io/ConsoleIO.java [hunk 1]
```diff
@@ -31,17 +32,27 @@ private ConsoleIO() {
...
+   * @see StreamingTransformTranslator#print()
    */
   public static final class Write {
 
     private Write() {
     }
 
-    public static <T> Unbound<T> out() {
-      return new Unbound<>(10);
+    private static final Integer DEFAULT_NUM = 10;
```
Comment (staslev) [inline runners/spark/src/main/java/org/apache/beam/runners/spark/io/ConsoleIO.java]: It would be nice to have all the fields positioned before any of the methods (ctor included).

File: runners/spark/src/main/java/org/apache/beam/runners/spark/io/ConsoleIO.java [hunk 2]
```diff
@@ -31,17 +32,27 @@ private ConsoleIO() {
 
   /**
    * Write on the console.
+   *
+   * @see StreamingTransformTranslator#print()
```
Comment (staslev) [inline runners/spark/src/main/java/org/apache/beam/runners/spark/io/ConsoleIO.java]: `StreamingTransformTranslator#print()` is a **private** method pertaining to the implementation of this transform, I would therefore avoid specifying it in the **public** javadoc aimed at telling users *what* this transform does, rather than *how*.

File: runners/spark/src/main/java/org/apache/beam/runners/spark/io/ConsoleIO.java [hunk 3]
```diff
@@ -31,17 +32,27 @@ private ConsoleIO() {
 
   /**
    * Write on the console.
+   *
```
Comment (staslev) [inline runners/spark/src/main/java/org/apache/beam/runners/spark/io/ConsoleIO.java]: It would suggest either keeping the old `Print to console.`  or going with  `Write to console.`

**Comment by joeyfezster** [comment]
@staslev changes made.

**Comment by staslev** [comment]
LGTM.

I'll merge this in once the tests pass.

**Comment by staslev** [comment]
Just noticed that something is wrong with the formatting.

This looks a bit off:

```java
      /**
       * Prints {@value #DEFAULT_NUM} elements from the {@link PCollection} to stdout.
       */
      public static <T> Unbound<T> out() {
      return new Unbound<>(DEFAULT_NUM);
    }
```

Could you please fix?

(You're welcome to use an [auto-formatter](https://plugins.jetbrains.com/plugin/8527-google-java-format))

**Comment by joeyfezster** [comment]
Code rebased and formatted

**Comment by joeyfezster** [comment]
Squashed

**Comment by joeyfezster** [comment]
Retest this please

File: runners/spark/src/main/java/org/apache/beam/runners/spark/io/ConsoleIO.java [hunk 4]
```diff
@@ -21,32 +21,31 @@
...
     }
 
+    /** Prints {@code num} elements from the {@link PCollection} to stdout. */
     public static <T> Unbound<T> out(int num) {
       return new Unbound<>(num);
     }
 
     /**
      * {@link PTransform} writing {@link PCollection} on the console.
+     *
```
Comment (staslev) [inline runners/spark/src/main/java/org/apache/beam/runners/spark/io/ConsoleIO.java]: Please use consistent terminology.
Some of javadoc says `Print to console`, some says `Print to stdout` and some says `writing to stdout`.

I suggest you pick one term and stick with it (either `write` or `print`, either `stdout` or `console`).

File: runners/spark/src/main/java/org/apache/beam/runners/spark/stateful/SparkGroupAlsoByWindowViaWindowSet.java [hunk 5]
```diff
@@ -204,6 +207,32 @@ public void outputWindowedValue(
         this.droppedDueToLateness = droppedDueToLateness;
       }
 
+      /**
```
Comment (staslev) [inline runners/spark/src/main/java/org/apache/beam/runners/spark/stateful/SparkGroupAlsoByWindowViaWindowSet.java]: This change is unrelated to this PR.

Did you rebase on the latest `master` branch?

File: runners/spark/src/main/java/org/apache/beam/runners/spark/io/ConsoleIO.java (hunk 4 above)
Comment (joeyfezster) [reply runners/spark/src/main/java/org/apache/beam/runners/spark/io/ConsoleIO.java]: I've changed all of those to `console`, and left the original discrepancy of `\*Print to console*\` and `\*Write to console*\`.

**Comment by joeyfezster** [comment]
@staslev  About the changes you saw before, they probably were committed between testing (twice because of the known fails).

Fresh rebase, updated to your latest remarks, and re-pushed.

**Comment by aaltay** [comment]
retest this please

**Comment by iemejia** [comment]
retest this please

**Comment by iemejia** [comment]
Run Spark ValidatesRunner

**Comment by iemejia** [comment]
Run Java Gradle PreCommit

**Comment by iemejia** [comment]
Thanks @joeyfezster and please excuse us for taking so long !
